azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/ExtensionProperties.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.domain;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableMap;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+import java.util.Map;
+
+@AutoValue
+public abstract class ExtensionProperties {
+
+   /**
+    * The publisher reference of the extension properties
+    */
+   public abstract String publisher();
+
+   /**
+    * The type reference of the extension properties
+    */
+   public abstract String type();
+
+   /**
+    * The typeHandlerVersion reference of the extension properties
+    */
+   public abstract String typeHandlerVersion();
+
+   /**
+    * The autoUpgradeMinorVersion reference of the extension properties
+    */
+   public abstract Boolean autoUpgradeMinorVersion();
+
+   /**
+    * The ExtensionProfileSettings of the extension properties
+    */
+   public abstract ExtensionProfileSettings settings();
+
+   /**
+    * The list of the protectedSettings of the extension properties
+    */
+   @Nullable
```

Comment:
You are enforcing presence and immutability. Remove this annotation.

Reply:
These @Nullable annotations are required to serialized the json response from the service (especially the when the value is not returned as sometimes there are not setup by the users)